src/app/components/challenge/challengesubmit/challengesubmit.component.ts

```diff
@@ -107,14 +107,21 @@ export class ChallengesubmitComponent implements OnInit {
   apiCall: any;
 
   /**
-   * Submissions remaining for the selected phase
+   * Selected phase submission conditions
+   * @param showSubmissionDetails show the selected phase submission details
+   * @param showClock when max submissions per day exceeded
+   * @param maxExceeded max submissions exceeded
+   * @param remainingSubmissions remaining submissions details
+   * @param maxExceededMessage message for max submissions exceeded
+   * @param clockMessage message for max submissions per day exceeded
    */
   selectedPhaseSubmissions = {
     showSubmissionDetails: false,
-    remainingSubmissions: {},
+    showClock: false,
     maxExceeded: false,
+    remainingSubmissions: {},
     maxExceededMessage: '',
-    message: ''
+    clockMessage: ''
```

Comment:
```suggestion
    clockMessage: String
```
Also, for the above one.

Reply:
Typecasting won't work for the dictionary but we should typecast as much as we should. I'll suggest you use **interfaces** instead (also in future PR's). https://stackoverflow.com/a/39387930/8681069